flybywiresim/aircraft
# fix(a32nx): Fix Chock Code for new GSX Update

Fixes N/A

## Summary of Changes

Changed Logic for Chock Removal in the XML Files.
With the new GSX Version 3.1.8 it is possible to request Pushback (pardon me ... Departure ... *sigh*) while Boarding is active. Depending on how that is triggered within GSX, the Variable `L:FSDT_GSX_DEPARTURE_STATE` can change to 4. In any Case, GSX will already Insert the By-Pass Pin, which will trigger the NWS DISC Memo.<br/>
In both Cases, the FBW Chocks will be removed as a Result. Which is rather unexpected for the User while Boarding is still ongoing and can lead to Issues with GSX when the Parking Brake is not set (because either one - Brakes or Chocks - needs to be set for GSX).<br/>
This PR fixes the Logic in the XML Files so that the Chock Removal is only triggered when the Departure Service is *really* active (5) and removes the Check for the NWS DISC Memo in the Rust Code.
Or put differently: I'm cleaning up (partly) my "Mess" from PR #8405 😅 
<br/>
If that Question should arise: No, that is not considered a Bug by GSX. Just got that confirmed by Umberto himself - for him everything is working as intended. (Apart from the Plane being frozen, which is not relevant to this PR).

## Screenshots (if necessary)



## References





## Additional context



Discord username (if different from GitHub): Fragtality

## Testing instructions

GSX Pro needed to test this PR.

Test-Case 1:
1. Ensure the FBW EFB Setting "Wheel Chocks" (Sim Options)  is *enabled*
2. Ensure the GSX Setting "Connect Pushback on boarding" is *disabled*
3. Load into Airport/Gate
4. Power Up Plane, Import OFP (if not configured for automatic Import)
5. Request Boarding (4) from the GSX Menu. If asked if Tug should be connected: Answer **No**
6. At any Time while Boarding is active, Request Departure/Pushback (5) from the GSX Menu

Test-Case 2:
1. Ensure the FBW EFB Setting "Wheel Chocks" (Sim Options)  is *enabled*
2. Load into Airport/Gate
3. Power Up Plane, Import OFP (if not configured for automatic Import)
5. Request Boarding (4) from the GSX Menu. If asked if Tug should be connected: Answer **Yes** (IF "Connect Pushback on boarding" is disabled. Else the Tug will be automatically attached by GSX.)

Result:
- Chocks should not be removed while Boarding is active
- (Only applies to Test-Case 1): They should be removed only after Boarding is finished and the Departure Service/Pushback becoming active


## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo** or **flybywire-aircraft-a380-842** download link at the bottom of the page
